commercetools.Sdk/commercetools.Sdk.Serialization/Extensions.cs

```diff
@@ -0,0 +1,52 @@
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace commercetools.Sdk.Serialization
+{
+    public static class Extensions
+    {
+        public static bool IsSetAttribute(this JToken valueProperty)
+        {
+            return valueProperty.IsArrayToken();
+        }
+        public static bool IsSetOfNestedAttribute(this JToken valueProperty)
+        {
+            var isArrayToken = valueProperty.IsArrayToken();
+            var array = valueProperty as JArray;
+            if (array == null || !isArrayToken || !array.Children().Any())
+                return false;
+
+            var isFirstItemNestedAttribute = array.First.IsNestedAttribute();
+            return isFirstItemNestedAttribute;
+        }
+        
+        public static bool IsNestedAttribute(this JToken valueProperty)
+        {
+            var isArrayToken = valueProperty.IsArrayToken();
+            var array = valueProperty as JArray;
+            if (array == null || !isArrayToken || !array.Children().Any())
+                return false;
+            
+            var allElementsContainsNameValueProps = true;
+            
+            foreach(var item in array.Children())
```

Comment:
it should be sufficient to check the first child to have name&value as properties as there is no other structure with these fields